bika/lims/browser/auditlog.py

```diff
@@ -0,0 +1,309 @@
+        # XXX: bad data, e.g. in AS Method field
+        elif value == "None":
+            value = _("Not set")
+        elif api.is_uid(value):
+            value = self.get_title_or_id_from_uid(value)
+        elif api.to_date(value):
+            value = self.to_localized_time(api.to_date(value))
+        elif isinstance(value, (list, tuple)):
+            value = map(self.process_value, value)
+            value = "  ".join(value)
+        return str(value)
+
+    def get_title_or_id_from_uid(self, uid):
+        """Returns the title or ID from the given UID
+        """
+        obj = api.get_object_by_uid(uid)
+        title_or_id = api.get_title(obj) or api.get_id(obj)
+        logger.info("get_title_or_id_from_uid: {} -> {}"
+                    .format(uid, title_or_id))
+        return title_or_id
+
+    def render_diff(self, diff):
+        """Render the diff template
+
+        :param diff: Dictionary of fieldname -> diffs
+        :returns: Rendered HTML template
+        """
+        return self.diff_template(self, diff=diff)
+
+    def get_snapshot_version(self, snapshot):
+        """Returns the version of the given snapshot
+        """
+        snapshots = self.get_snapshots()
+        return snapshots.index(snapshot)
+
+    def get_snapshot_metadata(self, snapshot):
+        """Returns the snapshot metadata
+        """
+        return snapshot.get("metadata", {})
+
+    def translate_state(self, s):
+        """Translate the given state string
+        """
+        if not isinstance(s, basestring):
+            return s
+        s = s.capitalize().replace("_", " ")
+        return t(_(s))
+
+    def folderitems(self):
+        """Generate folderitems for each version
+        """
+        items = []
+        # get the snapshots
+        snapshots = self.get_snapshots()
+        # reverse the order to get the most recent change first
+        snapshots = list(reversed(snapshots))
+        # set the total number of items
+        self.total = len(snapshots)
+        # slice a batch
+        batch = snapshots[self.limit_from:self.limit_from+self.pagesize]
+
+        for snapshot in batch:
+            item = self.make_empty_item(**snapshot)
+            # get the version of the snapshot
+            version = self.get_snapshot_version(snapshot)
+            # get the previous snapshot
+            prev_snapshot = self.get_snapshot_by_version(version-1)
+
+            diff = None
+            if prev_snapshot is not None:
+                diff = self.diff_snapshots(snapshot, prev_snapshot)
+
+            # Version
+            item["version"] = version
+
+            # get the metadata of the diff
+            metadata = self.get_snapshot_metadata(snapshot)
+
+            # Time
+            m_time = metadata.get("time")
+            item["time"] = self.to_localized_time(m_time)
+
+            # Modification Date
+            m_date = metadata.get("modified")
+            item["modified"] = self.to_localized_time(m_date)
```

Comment:
See my comment re "time" vs "modified" above. I would only display one of the two

Reply:
Yes, but unfortunately no modification takes place on wf transitions...